### PR TITLE
Unify eval formats

### PR DESCRIPTION
**Pull request by XiaLiPKU**
- parallelize detection eval
- unify format of eval_results
- rename: detect -> det; mot -> box_track

### CODE REVIEW

File: scalabel/eval/result.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+from typing import Callable, Dict, List, Optional
+
+import pandas as pd
+from mypy_extensions import KwArg, VarArg
+
+FORMATTER = Callable[[VarArg(object), KwArg(object)], str]
+
+
+class EvalResult:
```
Comment (fyu) [inline scalabel/eval/result.py:11]:
```suggestion
class Result:
```

**Review by fyu — DISMISSED** [review]
You can define different `Result` subclasses for different tasks to get nice print formatting.

File: scalabel/eval/mot.py [hunk 2]
```diff
@@ -313,12 +364,11 @@ def evaluate_track(
         nproc: processes number for loading files
 
     Returns:
-        dict: CLEAR MOT metric scores
+        BoxTrackResult: rendered eval results.
     """
     logger.info("Tracking evaluation with CLEAR MOT metrics.")
```
Comment (tobiasfshr) [inline scalabel/eval/mot.py:369]: this logger should be argument to the function, s.t. the user can decide where / if to print

File: scalabel/eval/mot.py [hunk 3]
```diff
@@ -350,24 +399,30 @@ def evaluate_track(
...
+            )
     else:
-        summaries = [
-            evaluate_single_class(name, acc) for name, acc in zip(names, accs)
+        flat_dicts = [
+            evaluate_single_class(names, accs)
+            for names, accs in zip(metric_names, class_accs)
         ]
 
     logger.info("rendering...")
```
Comment (tobiasfshr) [inline scalabel/eval/mot.py:418]: here we are not rendering anymore, we can delete this line

File: scalabel/eval/result.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+ScoresList = List[Scores]
+AVERAGE = "AVERAGE"
+OVERALL = "OVERALL"
+
+
+class Result(BaseModel):
+    """The base class for bdd100k evaluation results.
+
+    Each data field corresponds to a evluation metric. The value for each
+    metric is a dict that maps the category names to scores.
```
Comment (tobiasfshr) [inline scalabel/eval/result.py:22]: is now a list of dicts right?

File: scalabel/eval/result.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    _row_breaks: List[int] = PrivateAttr([])
+
+    def __init__(self, **data: Union[int, float, ScoresList]) -> None:
+        """Check the input structure and initiliaze the model."""
+        data_check: Dict[str, ScoresList] = {
+            metric: cont
+            for metric, cont in data.items()
+            if isinstance(cont, list)
+        }
+        ref_scores_list = data_check[list(data_check.keys())[0]]
```
Comment (tobiasfshr) [inline scalabel/eval/result.py:49]: if we check this here, we should add a remark to the doc that all keys need to be the same for the different metrics

File: scalabel/eval/result.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+            dict[str, int | float]: returned summary of the result
+        """
+        summary_dict: Dict[str, Union[int, float]] = dict()
+        for metric, scores_list in self.dict(
+            include=include, exclude=exclude  # type: ignore
+        ).items():
+            if not isinstance(scores_list, list):
+                summary_dict[metric] = scores_list
+            else:
+                summary_dict[metric] = scores_list[-1].get(
```
Comment (tobiasfshr) [inline scalabel/eval/result.py:164]: why still work with OVERALL / AVERAGE here? We can just print all metrics in the top-level of the hierarchy? Also please specify how the List reflects the hierarchy (bottom-up or top-down?)

File: scalabel/eval/mot.py [hunk 7]
```diff
@@ -42,6 +41,44 @@
     "mostly_lost": "ML",
     "num_fragmentations": "FM",
 }
+METRIC_TO_AVERAGE = set(["IDF1", "MOTA", "MOTP"])
+
+
+class BoxTrackResult(Result):
+    """The class for bounding box tracking evaluation results."""
+
+    mMOTA: float
```
Comment (tobiasfshr) [inline scalabel/eval/mot.py:50]: why do we still have mMOTA etc? This should correspond to the top-level category AVERAGE? So why store it separately again?

File: scalabel/eval/result.py [hunk 8]
```diff
@@ -123,6 +133,7 @@ def table(
             formatters = self._formatters
 
         summary = data_frame.to_string(formatters=formatters)
+        summary = summary.replace("NaN", " - ")
```
Comment (XiaLiPKU) [inline scalabel/eval/result.py:136]: NaN will not be printed now.
Replaced by '-'

**Review by tobiasfshr — APPROVED** [review]

**Pull request merged**